src/commands/createFunction/openAPISteps/OpenAPICreateStep.ts

```diff
@@ -3,109 +3,111 @@
-                break;
         }
 
-        args.push('--generate-metadata:false');
+        args.push(`--${pluginRunPrefix}-${pluginRun}`);
+        args.push(`--input-file:${cpUtils.wrapArgInQuotes(uri.fsPath)}`);
         args.push(`--output-folder:${cpUtils.wrapArgInQuotes(wizardContext.projectPath)}`);
+        args.push('--generate-metadata:false');
+        args.push(`--version:3.0.6320`);
 
         ext.outputChannel.appendLog(localize('statutoryWarning', 'Using the plugin could overwrite your custom changes to the functions.\nIf autorest fails, you can run the script on your command-line, or try resetting autorest (autorest --reset) and try again.'));
-        const title: string = localize('generatingFunctions', 'Generating from OpenAPI...Check [output window](command:{0}) for status.', ext.prefix + '.showOutputChannel');
+        ext.outputChannel.appendLog(localize('urlsForIssues', 'If there are any issues, please visit https://aka.ms/stencil for usage information or report issues at https://aka.ms/stencil/issues.'));
+        const title: string = localize('generatingFunctions', 'Generating from OpenAPI Specification...Check [output window](command:{0}) for status.', ext.prefix + '.showOutputChannel');
 
         await window.withProgress({ location: ProgressLocation.Notification, title }, async () => {
-            await cpUtils.executeCommand(ext.outputChannel, undefined, 'autorest', ...args);
+            try {
+                await cpUtils.executeCommand(ext.outputChannel, undefined, 'autorest', ...args);
+            } catch (error) {
+                const message: string = localize('autorestNotRunSuccessfully', `Failed to run "autorest.${pluginRunPrefix}-${pluginRun}".`);
+                if (!wizardContext.errorHandling.suppressDisplay) {
+                    window.showErrorMessage(message, DialogResponses.reportAnIssue).then(async result => {
+                        if (result === DialogResponses.reportAnIssue) {
+                            await openUrl('https://aka.ms/stencil/issues');
+                        }
+                    });
+                    wizardContext.errorHandling.suppressDisplay = true;
+                }
+
+                throw new Error(message);
+            }
         });
 
-        if (wizardContext.language === ProjectLanguage.TypeScript) {
-            // Change the package.json only when the autorest successfully ran.
-            await addAutorestSpecificTypescriptDependencies(wizardContext);
+        switch (wizardContext.language) {
+            case ProjectLanguage.TypeScript:
+                await openApiUtils.addAutorestSpecificTypescriptDependencies(wizardContext);
+                break;
+            case ProjectLanguage.Python:
+                await openApiUtils.addAutorestSpecificPythonDependencies(wizardContext);
+                break;
+            default:
+                break;
         }
     }
 
     public shouldExecute(): boolean {
         return true;
     }
 }
-
```

Comment:
Seems like you did some decently large refactoring and feature changes unrelated to my comments. In the future, please refrain from changes like that while a PR is in progress. Multiple smaller PRs will get reviewed/approved/merged much faster than one larger/evolving PR.